### ISSUE TITLE
if add a small Area2D in a big Area2D by code  ,the big Area2D can't detect the small Area2D in 3.4

### ISSUE THREAD

**Issue by colorworlds** [godotengine/godot#54749]
### Godot version

3.4.stable

### System information

win7, gles2

### Issue description

if add a small Area2D in a big Area2D **by code** ,the big Area2D can't detect the small Area2D

### Steps to reproduce

There are three situations：
1.Already has a node tree:
```
-Node2D:
-- BigArea
-- SmallArea
```
BigArea can detect SmallArea.

2.add a SmallArea by code,like :
```
var small_area = SmallArea.instance()
add_child(small_area)
small_area.position = big_area.position
```
BigArea can't detect SmallArea.

3.if the SamllArea is moving,it can work,it like:
```
small_area.position = Vector2.ZERO
func _process(delta: float) -> void:
	small_area.position += $BigArea2D.position /100.0
```

See the following project for details.

### Minimal reproduction project

[AreaBug.zip](https://github.com/godotengine/godot/files/7493972/AreaBug.zip)

**Comment by Giwayume** [comment]
I don't know how far this extends, my game loads entire rooms in a background thread and places them in the scene. I've been seeing the same issue where some Area2D just don't detect collisions in 3.4.stable.

I can set up a scenario where I can consistently reproduce this bug in my game in 3.4.stable, but if I swap back to the older 3.3.3.stable build the bug no longer exists. So it was introduced somewhere in that time-frame.

**Comment by Calinou** [comment]
@colorworlds @Giwayume Can you reproduce this in any of the [3.4 betas and RCs](https://downloads.tuxfamily.org/godotengine/3.4/) to determine when the regression started?

**Comment by Giwayume** [comment]
3.4.rc1 appears to be the last build where it worked correctly.

**Comment by Giwayume** [comment]
This could be tied to any KinematicBody2D changes as well, in my case it's Area2D detecting the player's KinematicBody2D.

When I press down on the keyboard to make the player crouch, I have it swapping out collision shapes (taller shape when standing, shorter when crouching). Both collision shapes are in the tree at the start of the game, it's just disabling one and enabling the other. At that point in time, the Area2D starts to detect the player again, no matter which collision shape is used (if they're crouching or standing). 

So perhaps it could be tied to initialization code of KinematicBody2D, and disabling/enabling the shapes fixes whatever the issue was that was missed during initialization.

**Comment by Giwayume** [comment]
Yeah, I've found even more problems with this, just instancing a scene in the tree that has a big area inside of it, that area isn't detecting pre-existing areas. That's broken in 3.4.rc1, whereas it works in 3.3. There seems to be a myriad of different issues with area to area detection.

**Comment by MikeFP** [comment]
Maybe this is related? #39765

**Comment by colorworlds** [comment]
邮件已查收~ ( *^o^* ) つロ 乾杯~
